### ISSUE TITLE
Unable to contact MacBook (OS X 10.6.8)

### ISSUE THREAD

**Issue by cpcallen** [prey/prey-node-client#126]
Having installed and [(eventually)](https://github.com/prey/prey-node-client/issues/125) configured Prey 1.3.9 on my MacBook Pro (mid 2010, OS X 10.6.8), I went to the devices panel on the Prey website (https://panel.preyproject.com/devices), where I noted that the device shows up but is listed as unavailable.

From the website it looks like the device is doing periodic check-ins in every 20 minutes or so, but clicking "update location" times out after a while with the error "Unable to get the devices location on time.", and attempting to send a message or set the alarm results in the message "Actions will be executed once the device reconnects to the Prey servers."

I have checked in System Preferences > Security > Firewall that "node" is permitted to accept incoming connections.

At the risk of conflating two separate issues, I also note that at least two attempts to sound an alarm and one to display a message resulted in no sound being produced (except for the harp and thump noises from the Prey website tab in my browser) or message even after an interval of > 20 minutes.

Here is the output of ```prey config log```:

```
  ████████████████ ████████████████ ████████████████ ██            ██
  ██            ██ ██               ██            ██ ██            ██
  ██            ██ ██               ████████████████ ██            ██
  ██            ██ ██               ██               ██            ██
  ████████████████ ██               ████████████████ ████████████████
  ██                                                               ██
  ██                                                 ████████████████

  PREY 1.3.9 spreads its wings!
  Current time: Tue Apr 28 2015 13:42:13 GMT+0100 (BST)
  Running with PID 62 as prey over Node.js v0.10.36 on a x64 mac system (10.8.0)

     info Tue, 28 Apr 2015 12:42:14 GMT [triggers] Watching: connection, network, power
     info Tue, 28 Apr 2015 12:42:14 GMT [actions] Starting trigger: connection
     info Tue, 28 Apr 2015 12:42:14 GMT [actions] Running: connection
     info Tue, 28 Apr 2015 12:42:14 GMT [actions] Starting trigger: network
     info Tue, 28 Apr 2015 12:42:14 GMT [actions] Starting trigger: power
     info Tue, 28 Apr 2015 12:42:18 GMT [actions] Running: network
     info Tue, 28 Apr 2015 12:42:18 GMT [actions] Running: power
   notice Tue, 28 Apr 2015 12:42:18 GMT [control-panel] Checking for existing port mappings...
     info Tue, 28 Apr 2015 12:42:18 GMT Plugin loaded: control-panel
     info Tue, 28 Apr 2015 12:42:18 GMT 1 plugins loaded with 0 errors.
     info Tue, 28 Apr 2015 12:42:18 GMT Initialized.
     info Tue, 28 Apr 2015 12:42:18 GMT [control-panel] Queuing check-ins every 0.5 minutes.
     warn Tue, 28 Apr 2015 12:42:49 GMT [control-panel] Got 1 commands.
     info Tue, 28 Apr 2015 12:42:53 GMT [reports] Gathering status report.
     info Tue, 28 Apr 2015 12:42:57 GMT [reports] Report gathered: status
     info Tue, 28 Apr 2015 12:42:57 GMT [control-panel] Posting data
     info Tue, 28 Apr 2015 12:42:58 GMT [control-panel] Got 200 response: OK
     warn Tue, 28 Apr 2015 12:45:51 GMT [control-panel] Got 1 commands.
     info Tue, 28 Apr 2015 12:45:55 GMT [reports] Gathering status report.
     info Tue, 28 Apr 2015 12:45:59 GMT [reports] Report gathered: status
     info Tue, 28 Apr 2015 12:45:59 GMT [control-panel] Posting data
     info Tue, 28 Apr 2015 12:46:03 GMT [control-panel] Got 200 response: OK
     warn Tue, 28 Apr 2015 12:46:22 GMT [control-panel] Got 1 commands.
     info Tue, 28 Apr 2015 12:46:22 GMT [actions] Starting action: alert
     info Tue, 28 Apr 2015 12:46:22 GMT [control-panel] Posting response
     info Tue, 28 Apr 2015 12:46:22 GMT [actions] Running: alert
     info Tue, 28 Apr 2015 12:46:22 GMT [actions] Stopped: alert
     info Tue, 28 Apr 2015 12:46:23 GMT [control-panel] Got 200 response: OK
     info Tue, 28 Apr 2015 12:46:23 GMT [control-panel] Posting response
     info Tue, 28 Apr 2015 12:46:24 GMT [control-panel] Got 200 response: OK
```

I should note that my machine rebooted while I was preparing this report (battery expended -> safe sleep -> wake apparently unsuccessful -> reboot), so log is missing some of the alarm/message attempts, but as of now (13:52 GMT) there have been no log entries, alarms or messages in the last hour despite the website control panel showing a message pending, device last seen a minute ago and device location last updated 3 minutes ago.

**Comment by lemavri** [comment]
Could you please run `ps aux | grep prx | grep grep -v` in the terminal and let me know what the output is?

If you get no output, could you please run 

```sudo launchctl load /Library/LaunchDaemons/com.prey.agent.plist```

and repeat the `ps aux` command mentioned above?

**Comment by lemavri** [comment]
> From the website it looks like the device is doing periodic check-ins in every 20 minutes or so

**Comment by cpcallen** [comment]
At the present time:

```
$ ps aux | grep prx | grep grep -v
prey        62   0.0  0.7  3073052  61576   ??  Ss    1:42pm   0:02.86 prx
```
(Local time is BST == GMT+1)

I see note that user prey is also running ```scutil```.

At the time I submitted this issue, my laptop Ansel was shown as unavailable (with grey dot), and was generally last seen 1 <= n < 20ish minutes ago, hence my suspecting periodic connections.  It would respond to location requests only after some large (5-15 minute) delay.

It seems that it is now connected; here's the screenshot of the website control panel for my laptop as of a few minutes ago:
![screen shot 2015-04-28 at 15 18 50](https://cloud.githubusercontent.com/assets/5603958/7371762/c80d9038-edba-11e4-836c-dad480c47fa7.png)

Additional log lines since last report:
```
     warn Tue, 28 Apr 2015 13:49:25 GMT [control-panel] Got 1 commands.
     info Tue, 28 Apr 2015 13:49:28 GMT [reports] Gathering status report.
     info Tue, 28 Apr 2015 13:49:32 GMT [reports] Report gathered: status
     info Tue, 28 Apr 2015 13:49:32 GMT [control-panel] Posting data
     info Tue, 28 Apr 2015 13:49:33 GMT [control-panel] Got 200 response: OK
     warn Tue, 28 Apr 2015 14:17:53 GMT [control-panel] Got 1 commands.
     info Tue, 28 Apr 2015 14:17:57 GMT [reports] Gathering status report.
     info Tue, 28 Apr 2015 14:18:01 GMT [reports] Report gathered: status
     info Tue, 28 Apr 2015 14:18:01 GMT [control-panel] Posting data
     info Tue, 28 Apr 2015 14:18:05 GMT [control-panel] Got 200 response: OK
     warn Tue, 28 Apr 2015 14:23:59 GMT [control-panel] Got 1 commands.
     info Tue, 28 Apr 2015 14:24:02 GMT [reports] Gathering status report.
     info Tue, 28 Apr 2015 14:24:06 GMT [reports] Report gathered: status
     info Tue, 28 Apr 2015 14:24:06 GMT [control-panel] Posting data
     info Tue, 28 Apr 2015 14:24:13 GMT [control-panel] Got 200 response: OK
```
(There is nothing in the log between 12:46:25 GMT and 13:49:25 GMT.)

I note the message is shown as still pending (and has been for more than an hour), despite the device being listed as available.

I've now tried tried cancelling the delayed message request and resubmitting message / alarm requests.  These appeared to go through successfully within a few seconds, and generated the following log lines:

```
     warn Tue, 28 Apr 2015 14:37:42 GMT [control-panel] Got 1 commands.
     info Tue, 28 Apr 2015 14:37:42 GMT [actions] Starting action: alert
     info Tue, 28 Apr 2015 14:37:42 GMT [control-panel] Posting response
     info Tue, 28 Apr 2015 14:37:42 GMT [actions] Running: alert
     info Tue, 28 Apr 2015 14:37:43 GMT [actions] Stopped: alert
     info Tue, 28 Apr 2015 14:37:44 GMT [control-panel] Posting response
     info Tue, 28 Apr 2015 14:37:51 GMT [control-panel] Got 200 response: OK
     info Tue, 28 Apr 2015 14:37:51 GMT [control-panel] Got 200 response: OK
     warn Tue, 28 Apr 2015 14:39:44 GMT [control-panel] Got 1 commands.
     info Tue, 28 Apr 2015 14:39:44 GMT [actions] Starting action: alarm
     info Tue, 28 Apr 2015 14:39:44 GMT [control-panel] Posting response
     info Tue, 28 Apr 2015 14:39:44 GMT [actions] Running: alarm
     info Tue, 28 Apr 2015 14:39:45 GMT [actions] Stopped: alarm
     info Tue, 28 Apr 2015 14:39:46 GMT [control-panel] Posting response
     info Tue, 28 Apr 2015 14:39:48 GMT [control-panel] Got 200 response: OK
     info Tue, 28 Apr 2015 14:39:50 GMT [control-panel] Got 200 response: OK
     warn Tue, 28 Apr 2015 14:40:15 GMT [control-panel] Got 1 commands.
     info Tue, 28 Apr 2015 14:40:15 GMT [actions] Starting action: alarm
     info Tue, 28 Apr 2015 14:40:15 GMT [control-panel] Posting response
     info Tue, 28 Apr 2015 14:40:15 GMT [actions] Running: alarm
     info Tue, 28 Apr 2015 14:40:15 GMT [actions] Stopped: alarm
     info Tue, 28 Apr 2015 14:40:16 GMT [control-panel] Posting response
     info Tue, 28 Apr 2015 14:40:19 GMT [control-panel] Got 200 response: OK
     info Tue, 28 Apr 2015 14:40:20 GMT [control-panel] Got 200 response: OK
```

Still no messages seen or alarms heard, however.

So this now appears to be a separate issue from the one I originally reported.

**Comment by lemavri** [comment]
The alert makes use of Python and it could be related to the other issue you reported. Is the alarm being executed correctly?

**Comment by cpcallen** [comment]
If by "alarm being executed properly" you mean "the computer produces some
sound", then the answer is positively not.

(The only sound I have so far heard are the little chimes that my web
browser produces when using the Prey website; they are neither loud nor
more than a second or so in length.)

**Comment by lemavri** [comment]
I'm sorry for the delay @cpcallen. Could you please run the following command in your terminal? (make sure not to be wearing headphones 😜)

 `afplay /usr/lib/prey/current/lib/agent/actions/alarm/lib/alarm.mp3`

**Comment by cpcallen** [comment]
> I'm sorry for the delay @cpcallen. Could you please run the following
> command in your terminal? (make sure not to be wearing headphones )
>
> afplay /usr/lib/prey/current/lib/agent/actions/alarm/lib/alarm.mp3

This makes an annoying beeping noise (provided my system volume is not muted).


Christopher

**Comment by camilaburgos** [comment]
Hi, @cpcallen. Can you please update the version your computer's running to our latest release (**1.6.1**)? Does this version solve the problems or do they keep happening?

**Comment by cpcallen** [comment]
> Hi, @cpcallen. Can you please update the version your computer's
> running to our latest release (1.6.1)? Does this version solve the
> problems or do they keep happening?

It appears that Prey has already updated itself to 1.6.1 - or at any
rate, /usr/local/lib/prey/current exists and points to
/usr/local/lib/prey/versions/1.6.1 which also exists and is non-empty.

Here is the behaviour I observed when I went to
https://panel.preyproject.com/devices/792pul just now:

- It incorrectly reports my machine as "Apple MacBook, OS X Snow
Leopard (10.6.8)
Prey: 1.5.0"; in fact it is running OS X 10.11.5 and if not Prey 1.6.1
then the only other installed version, which is 1.5.1.

- It shows location as "about a year ago"; clicking on "Update
Location" has no effect (location update fails).

- The "Send Message" and "Sound Alarm" functions both work (i.e.,
message appears and sound is heard) but the Prey UI continues to show
"This action is pending" even once message/alarm complete; eventually
it reports "Actions will be executed once the device reconnects to the
Prey servers."

I decided to do a complete uninstall/reinstall, in case there is some
corruption/misconfiguration issue which is causing problems.  I
followed the instructions at
http://help.preyproject.com/article/190-how-to-uninstall-prey, then
re-added the device via the interface; this downloaded
prey-mac-1.6.1-x86.pkg which I installed and configured.

After doing this I now have a new entry for my device
(https://panel.preyproject.com/devices/ca6891), and everything appears
to be working properly:

- OS and Prey version reported correctly.

- Location determined correctly.

- "Send Message" and "Sound Alarm" work properly (as before) AND are
shown in the control panel as having completed successfully (unlike
before).

So an uninstall/reinstall fixed all outstanding issues.

It is kind of bad that Prey managed to get itself in a state where,
despite apparently successful auto-updates, it was not actually
working fully.  Perhaps I should file a separate bug about this - only
I probably destroyed the evidence necessary to reproduce it when
uninstalling/reinstalling just now, sadly...


Thanks for a good, useful and now working tool!

**Comment by camilaburgos** [comment]
@cpcallen That's fantastic! I'm so glad it did the trick :)
We migrated from the Batch client to Node (0.6.4 to 1.x.x version), so it's no surprise that the automatic upgrade got a little difficult and Prey just stopped working. From now on, the app should be automatically updated every time we release a new version, so you can rest easy!
